src/otx/v2/adapters/torch/mmengine/modules/hooks/adaptive_training_hook.py

```diff
@@ -4,11 +4,12 @@
 #
 
 import math
-from typing import Optional
+from typing import Optional, Sequence, Dict
 
 from mmengine.hooks import CheckpointHook, Hook
 from mmengine.registry import HOOKS
-from mmengine.runner import Runner
+from mmengine.runner import Runner, EpochBasedTrainLoop
+from mmengine.hooks import ParamSchedulerHook
 
 from otx.v2.adapters.torch.mmengine.modules.hooks.early_stopping_hook import (
     EarlyStoppingHook,
```

Comment:
Maybe we can removed our customised EarlyStoppingHook as MMEngine has it now: https://github.com/open-mmlab/mmengine/pull/739

Reply:
I agree